src/inference/src/compilation_context.cpp

```diff
@@ -120,7 +120,54 @@ std::string NetworkCompilationContext::computeHash(const CNNNetwork& network,
         seed = hash_combine(seed, as_int32_t(info->getPrecision()));
         seed = hash_combine(seed, as_int32_t(info->getLayout()));
     }
+    return std::to_string(seed);
+}
+
+std::string NetworkCompilationContext::computeFasterHash(const CNNNetwork& network,
+                                                         const std::map<std::string, std::string>& compileOptions) {
+    OV_ITT_SCOPE(FIRST_INFERENCE, itt::domains::IE_LT, "NetworkCompilationContext::computeFasterHash - CNN");
+
+    IE_ASSERT(network.getFunction());
+
+    uint64_t seed = 0;
+    // 1. Calculate hash on function
+    CNNNetwork net(network);
```

Comment:
Why we use CNNNetwork here if it is only used to get function from it?
If you want to make a copy, so please use `ov::clone_function(*network.getFunction());`

Reply:
Making a copy brings a big overhead. This CNNNetwork object is created just because we can't run 'pass' on `shared_ptr<const ov::Model>`. Alternative would be to use a 'const_pointer_cast' which looks even worse.
Creating CNNNetwork this way causes just a copy of shared_ptr<Model> and it is cheap. We don't modify model here when calculating hash